pynetdicom/events.py

```diff
@@ -420,6 +426,33 @@ def dataset(self):
         )
         return self._get_dataset("DataSet", msg)
 
+    @property
+    def dataset_file(self):
+        dataset_file = self.request._dataset_file
+
+        if not dataset_file:
+            raise RuntimeError("No dataset file available")
+
+        if self._did_prepare_dataset_file:
+            return dataset_file
+
+        # Prepend File Meta Information
+        with open(
+            f"{dataset_file}.meta", 'w+b'
+        ) as meta_file, open(
+            dataset_file, 'rb'
+        ) as data_set_file:
+            meta_file.write(b'\x00' * 128)
+            meta_file.write(b'DICM')
+            write_file_meta_info(meta_file, self.file_meta)
+            copyfileobj(data_set_file, meta_file)
```

Comment:
Performance-wise, it would be preferable to write the file_meta in `decode_message` but I didn't see any easy way to access the accepted Presentation Context there (required to write TransferSyntaxUID). This is sort of nice though since we already have `file_meta` available on the event.

Reply:
You can change the function signature to `DIMSEMessage.decode_msg(primitive, assoc)`, then access the accepted context using `assoc._accepted_cx[context_id]`.